widget/entry.go

```diff
@@ -1158,17 +1158,23 @@ func (e *Entry) truncatePosition(row, col int) (int, int) {
 
 func (e *Entry) updateMousePointer(p fyne.Position, rightClick bool) {
 	row, col := e.getRowCol(p)
-	e.setFieldsAndRefresh(func() {
-		if !rightClick || rightClick && !e.selecting {
-			e.CursorRow = row
-			e.CursorColumn = col
-		}
+	e.propertyLock.Lock()
 
-		if !e.selecting {
-			e.selectRow = row
-			e.selectColumn = col
-		}
-	})
+	if !rightClick || rightClick && !e.selecting {
```

Comment:
You should be able to simplify the Boolean logic slightly:
```suggestion
	if !rightClick || !e.selecting {
```
As you can see, the truth tables are the same:
![image](https://user-images.githubusercontent.com/25466657/204126716-75c61df8-a477-4cb8-a224-e5dbec7088bc.jpeg)
![image](https://user-images.githubusercontent.com/25466657/204126721-fe8e5517-1d26-4179-9d69-3f930fc7ae57.jpeg)

Reply:
Good point. Missed that as it was just an indent change. fixing